application/controllers/ContactController.php

```diff
@@ -40,21 +40,17 @@ public function indexAction()
             ->populate($contact)
             ->on(ContactForm::ON_SUCCESS, function (ContactForm $form) {
                 $form->addOrUpdateContact();
-                /** @var FieldsetElement $contactElement */
-                $contactElement = $form->getElement('contact');
                 Notification::success(sprintf(
                     t('Contact "%s" has successfully been saved'),
-                    $contactElement->getValue('full_name')
+                    $form->getValue('contact')['full_name']
```

Comment:
There's absolutely no question that the previous case in the remove handler was wrong, but why is this changed?

My understanding is, and phpstan seems to agree, that this is now even more relying on implementation details than before. Yes, the full name is required, so the key always exists. But there's no way to express, even in the future, that the `contact` is a fieldset and has a field called `full_name` which holds a string.

I'd rather keep this as it was, and apply the same handling in the remove handler.

Reply:
That was my suggestion so that both pieces of code look the same. But yes, I can understand your point.